test/test_fit.py

```diff
@@ -278,6 +248,18 @@ def test_fit_gpytorch_model_singular(self):
                 fit_gpytorch_model(
                     mll, options={"disp": False, "maxiter": 1}, max_retries=1
                 )
+            # ensure we catch NotPSDErrors
+            with mock.patch.object(SingleTaskGP, "__call__", side_effect=NotPSDError):
+                mll = self._getModel()
+                with self.assertLogs(level="DEBUG") as logs:
+                    fit_gpytorch_model(mll, max_retries=2)
+                self.assertEqual(
+                    logs.output,
+                    [
+                        "DEBUG:root:Fitting failed on try 1 due to a NotPSDError.",
+                        "DEBUG:root:Fitting failed on try 2 due to a NotPSDError.",
+                    ],
+                )
```

Comment:
The equality assertion here could be quite brittle if something benign changed (e.g. the logging prefix, or if other stuff is logged on the gpytorch end). Maybe we can just assert that "Fitting failed on try 1 due to a NotPSDError." and "Fitting failed on try 2 due to a NotPSDError." are both contained in the logs (rather than doing an exact log match)?